### ISSUE TITLE
Index issue: Disabled variant details are shown on product cards

### ISSUE THREAD

**Issue by erinemay** [iFixit/react-commerce#220]
### Description
Product cards are displaying disabled variant information. 
Product: https://www.ifixit.com/Store/iPhone/iPhone-4S-Battery/IF115-005?o=7
Product card on https://react-commerce.vercel.app/store/parts?p=5: 
![image](https://user-images.githubusercontent.com/72166715/159079367-a5d587fa-2e4a-48cc-8b3b-93d21bc8393c.png)

We're showing the price, and variant information for IF115-005-2 instead of the default product IF115-005-7
Akeneo:
![image](https://user-images.githubusercontent.com/72166715/159079541-1ca89d71-474d-439e-9dd5-bb7e13101e7f.png)

### Context
https://github.com/iFixit/react-commerce/issues/122 and https://github.com/iFixit/react-commerce/issues/219 appear to be symptoms of this.

**Comment by erinemay** [comment]
CC: @sterlinghirsh

**Comment by sterlinghirsh** [comment]
OK I looked at the index a little and I have a few ideas. First, we should be including a filter in our search to say public=1. As of a couple months ago ish, there are disabled products in our index that are intended to be searchable by admins, but they are all marked public=0. Not sure if there's anything we need to change about indexing to make sure public=1 is always used on this page.

I noticed we also have a ranking of results that uses the has_image and orderby fields as well as one or two others. This may be moot after filtering out the public=0 results, but we should probably changing the indexing to sort public=0 to the end in any case.

Looks like we already exclude disabled products from the group min/max prices. group_min_price and group_max_price are what we use to show a price range when variants have different prices. Since we're showing products and not variants, it seems like we'll often have to show a range or something like "From $19.99"

**Comment by dhmacs** [comment]
@sterlinghirsh 

> First, we should be including a filter in our search to say public=1.

Can you add it as a facet so I can open a PR for this?

> Looks like we already exclude disabled products from the group min/max prices. group_min_price and group_max_price are what we use to show a price range when variants have different prices. 

I have a few questions here:

- Are `group_max_price` and `group_min_price` always defined, even for products with a single variant? Can I rely on those?
- How are we going to deal with discounted prices? Right now we're using `compare_at_price`, but it seems that we should include in the index `group_min_compare_at_price` and `group_max_compare_at_price` too, right?

> Since we're showing products and not variants, it seems like we'll often have to show a range or something like "From $19.99"

Good idea, let me know if I should go ahead and open a PR for this too

**Comment by sterlinghirsh** [comment]
We an add it as a facet, though we don't currently have it as a facet and we still filter on it.

> group_max_price

These are defined for enabled variants. If a product has only disabled variants, it will be null.

I don't really know what we want to do with group compare at price. Seems like we should only show the crossed out one if it's higher than group_max_price. @jeffsnyder might want to weigh in.

**Comment by jeffsnyder** [comment]
I think we need both a min and max compare at. 99% of the time we'll be discounting the higher priced item, but we should build this to handle both. We should just show two price ranges: 

$14.99 - $54.99    ~$29.99 - $54.99~

But we'll also need to address when we drop a fix kit to equal the price of the part only (this happens sometimes): 

$24.99 ~$24.99 - $29.99~

**Comment by sterlinghirsh** [comment]
Have you seen other sites show a crossed out range of prices? Seems kind of weird to me.

**Comment by jeffsnyder** [comment]
@sterlinghirsh Yeah, you're right. Our current standard on Product List Pages is `From [lowest price]`. But we do show price ranges in search. The price range doesn't play nice with sale pricing. I think BackMarket does it right: 

![image](https://user-images.githubusercontent.com/7053450/159382935-61536078-7b37-4259-8dff-0b59589eb354.png)

**Comment by jeffsnyder** [comment]
@dhmacs I like that plan. I can't think of of edge cases. 

> We can also add a boolean to the index named like discounted_options so that we can use to show a label "Discounted options available", telling the user that there are discounted variants.

I like this. I don't think I've seen something similar. The wording is going to matter a lot here. Maybe we should just add `SALE` flair to all Product Cards where one or more Variants in the Group are on sale? It might not be perfectly explanatory, but it gets the point across without much fuss.

**Comment by dhmacs** [comment]
Gotcha 👍 Then to implement this I need `group_min_compare_at_price` and `discounted_options` to be on the index

cc @sterlinghirsh

**Comment by sterlinghirsh** [comment]
- Only show a crossed-out compare_at_price if it's greater than the displayed price.
 - If customer is a pro, use the normal price as the compare_at_price if compare_at_price is null in the index.
 - When multiple variants are available, there are two options. 1. Show the current variant result only (no range, no "from). 2. Show "From [group_min_price]".
   - After discussing with @jeffsnyder and @lithobraking we decided to go with option 1. We'll ignore the fact that there are multiple variants. We decided that our common case of advertising fix kits would be too misleading if we show them with part-only prices, since this has previously lead to too many customer complaints.
